rustls/src/client/tls12.rs

```diff
@@ -1072,8 +1072,11 @@ impl State<ClientConnectionData> for ExpectTraffic {
             MessagePayload::ApplicationData(payload) => cx
                 .common
                 .take_received_plaintext(payload),
-            _ => {
-                return Err(inappropriate_message(&m, &[ContentType::ApplicationData]));
+            payload => {
```

Comment:
Here, you used `payload => .. &payload ..` whereas in other commits it seems you've mostly used `ref payload`. Why the distinction? IMO the former style is currently more idiomatic.

Reply:
> Here, you used payload => .. &payload .. whereas in other commits it seems you've mostly used ref payload. Why the distinction? IMO the former style is currently more idiomatic.

I pushed a new commit on top of the existing ones that makes them all consistent.